File: ParrotSpam.java
public class ParrotSpam {
    public static void main(String[] args) {
        int avail = 4000;
        int total = 0;

        for (String arg : args) {
            total += arg.length();
        }

        if (total == 0) {
            System.err.println("Nothing to repeat");
            System.exit(1);
        }

outer:
        while (true) {
            for (String arg : args) {
                if (avail < arg.length()) {
                    break outer;
                }
                System.out.print(arg);
                avail -= arg.length();
            }
        }
    }
}

File: parrot_spam.pl
use strict;
use warnings;
use v5.18;

my @parrots = grep { length $_ > 0 } @ARGV;
unless (@parrots) {
    print STDERR "Nothing to repeat\n";
    exit 1;
}

my $avail = 4000;

SPAM: while (1) {
    foreach my $p (@parrots) {
        $avail -= length $p;
        last SPAM if $avail < 0;
        print $p;
    }
}

print "\n" if -t STDIN;
